scripts/globals/pasteventwatcher.lua

```diff
@@ -0,0 +1,60 @@
+-----------------------------------
+-- Past Event Watcher NPCs
+-----------------------------------
+
+-----------------------------------
+
+dsp = dsp or {}
+dsp.pasteventwatcher = dsp.pasteventwatcher or {}
+
+dsp.pasteventwatcher.goblinFootprint = function(options)
+    -- Defaults options to 0 if not all of them have been passed.
+    for i=1,31, 1 do
+        if not options[i] then
+            options[i] = 0
+        end
+    end
+
+    -- 32 bits represents a cutscene being watched or not
+    -- The first bit defaults to 0 to show the "Another time maybe" choice to let user get out from the dialog
+    -- The first scene starts at bit2, and it has to be 0 if it becomes part of the list, otherwise it should be a 1.
+    -- If you need to know the correct order of the scenes use !cs <scene number> for the gobbie to list them all
+    local bit32 = (1-options[31])*2147483648
```

Comment:
I'm a little confused here. Where are these ever being used? You only ever pass in an options array with 4 elements. Are there missing NPCs that aren't implemented yet?

Reply:
Hi @nfaletra this PR is for the function logic to be scrutinized with 1 goblin before I start writing the other logic for all the other goblins as none has been scripted in DSP, have already tested it in other 3 areas but my logic looks horrible for the other ones but wanted to get the go ahead on this first before spending more time and having to rewrite it, since theres a ton of gobbies lol  😺 notes are on the 1st comment of PR (sorry if i wasn't clear, English is my second language)